spec/spec.yaml

```
@@ -3936,6 +3952,180 @@ paths:
+                  "foo": "bar"
+                }
+              }
+        "401":
+          $ref: "./responses.yaml#/responses/V4Generic401Response"
+        "404":
+          description: Not found
+          schema:
+            $ref: "./definitions.yaml#/definitions/V4GenericResponse"
+          examples:
+            application/json:
+              {
+                "code": "RESOURCE_NOT_FOUND",
+                "message": "The cluster with ID 'wqt' could not be found..."
+              }
+    put:
+      operationId: setClusterLabels
+      tags:
+        - cluster labels
+      summary: Set cluster labels
+      description: |
+        Set cluster labels. Keys and labels should adhere to
+        [Kubernetes labels syntax and character set](https://v1-16.docs.kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set).
+
+        Label changes should be written as a [JSON Merge Patch, RFC 7386](https://tools.ietf.org/html/rfc7386).
+
+        Changes to labels containing `giantswarm.io` is forbidden,
+        changes to label `release.giantswarm.io/version`
+        will be validated against available Giant Swarm releases.
+      parameters:
+        - $ref: "./parameters.yaml#/parameters/ClusterIdPathParameter"
+        - name: body
+          in: body
+          required: true
+          description: Labels to attach to this cluster.
+          schema:
+            $ref: "./definitions.yaml#/definitions/V5ClusterLabelsResponse"
+          x-examples:
+            application/json:
+              {
+                "labels": {
+                  "environment": "testing-2",
+                  "region": "north-west-2",
+                  "label-to-remove": null
+                }
+              }
+      responses:
+        "200":
+          description: Labels attached to this cluster after update
+          schema:
+            $ref: "./definitions.yaml#/definitions/V5ClusterLabelsResponse"
+          examples:
+            application/json:
+              {
+                "labels": {
+                  "environment": "testing-2",
+                  "region": "north-west-2"
+                }
+              }
+        "401":
+          $ref: "./responses.yaml#/responses/V4Generic401Response"
+        "404":
+          description: Not found
+          schema:
+            $ref: "./definitions.yaml#/definitions/V4GenericResponse"
+          examples:
+            application/json:
+              {
+                "code": "RESOURCE_NOT_FOUND",
+                "message": "The cluster with ID 'wqt' could not be found..."
+              }
+
+  /v5/clusters/by_label/:
+    post:
+      operationId: getV5ClustersByLabel
+      tags:
+        - clusters
+      summary: Get clusters by labels (v5)
+      description: |
+        This operation fetches a list of node pool clusters based
+        on a label selector.
+
+        The operation accepts label selectors analogue to `kubectl get -l`
```

```suggestion
        The operation accepts label selectors in the same way that `kubectl get -l` does
```

I try to make language in docs as simple as possible, even if it means a few more words, I think it ends up faster for most people to parse.